spec.md

```diff
@@ -0,0 +1,117 @@
+                - the funder must have a `token` balance of at least `amount`
+                - the factory must have a `token` allowance of at least `amount`
+        - `vesting_start`
+            - can be in the past
+        - `cliff_duration`
+            - must not exceed vesting duration
+    - actions
+        - create a new vesting escrow using `create_minimal_proxy_to`
+        - fund it by transferring tokens from `msg.sender` to the newly created `escrow`
+        - `initialize` the `escrow`
+        - log the creation parameters in `VestingEscrowCreated` event
+
+
+## Vesting Escrow
+
+- global considerations
+    - return an appropriate value or True from methods to make contract calls cheaper
+    - use `default_return_value=True` for token transfers
+
+- `__init__`
+    - actions
+        - template contract must be made defunct by storing `initialized` as True
+- `initialize`
+    - arguments
+        - token
+        - recipient
+        - amount
+        - start_time
+        - end_time
+        - cliff_duration
+        - open_claim
+        - admin
+    - constraints
+        - must not be initialized
+        - must be funded
+            - implicitly checked by only allowing creation from factory
+    - actions
+        - save the creation parameters to storage
+        - store `initialized` as True
+    - returns
+        - True to make the call from a contract cheaper
+- `claim`
+    - arguments
+        - `beneficiary: address = msg.sender` where to send funds to
+        - `amount: uint256 = max_value(uint256)` optionally claim a fixed amount
+    - constraints
+        - recipient can claim to any beneficiary
+        - if beneficiary is the recipient and `open_claim` is True, allow anyone to claim
+        - the `amount` is limited by the maximum amount claimable
+    - actions
+        - `total_claimed` is updated
+        - transfer tokens to beneficiary, use `default_return_value=True`
+        - log `Claim` with beneficiary and claimed amount
+    - returns
+        - amount claimed
+- `terminate` (previously `rug_pull`)
+    - arguments
+        - `time: uint256 = block.timestamp` optionally terminate at a date and clawback lower amount
+        - `beneficiary: address` where to send the clawed back amount
+            - one use case could be to reward the recipient early, another is to transfer to treasury and not have admin handle to funds
+    - constraints
+        - can only be called by `admin`
+        - time must be now or in the future
+        - time must not exceed vesting end time
+    - actions
+        - `disabled_at` is set to `time`
+        - admin is set to `empty(address)`
+        - the amount of tokens is determined as tokens still locked at `time`
+        - tokens are transferred to `admin`
+        - log `VestingTerminated(self.recipient, self.admin, rugged, time)`
+        - log `SetFree(self.admin)`
+- `set_free` (previously `set_admin`)
```

Comment:
usually i seen `renounce_admin` or `renounce_ownership` used elsewhere to clarify the intention of the func.

set_free may be understood as a setting called `free` at first glance.

Reply:
will change to `disown``